src/components/ForecastMainBanner.svelte

```diff
@@ -0,0 +1,112 @@
+<script>
+  import { Spinner } from 'flowbite-svelte';
+
+  import thermostat from '../assets/thermostat.svg';
+  import arrow from '../assets/arrow.svg';
+  import umbrella from '../assets/umbrella.svg';
+  import air from '../assets/air.svg';
+  import clock from '../assets/clock.svg';
+  import weatherCodes from '../assets/weather-interpretation-code-description.json';
+  import getDayPeriod from '../utils/dayPeriod';
+
+  export let weatherData = {};
+  const dayPeriod = getDayPeriod(weatherData.time);
+</script>
+
+<div class="p-5 md:p-10">
+  <div class="pt-2 p-4 md:pt-5 md:px-10 rounded-lg forecast-panel">
+    <h2 class="flex items-center font-medium text-sm md:text-lg">
+      <span>
+        <img class="h-4 md:h-6" src={clock} alt="clock" />
+      </span>
+      <span>&nbsp;Current conditions</span>
+    </h2>
+
+    {#if weatherData.isLoading}
+      <div class="flex justify-center items-center p-4 md: pt-5">
+        <Spinner />
+      </div>
+    {:else if weatherData.hasError}
+      <div class="flex justify-center items-center p-4 md: pt-5">
+        <p>No data received</p>
+      </div>
+    {:else}
+      <div class="flex justify-between items-center">
+        {#if weatherData.weatherCode !== null && weatherCodes[weatherData.weatherCode]?.[dayPeriod]?.image}
```

Comment:
just `weatherData.weatherCode` should be enough

Reply:
But weather code possibly could be 0. And zero in this case is valid value.

Reply:
Good point, disregard